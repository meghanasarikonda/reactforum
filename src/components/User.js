import React from 'react';
import PropTypes from 'prop-types';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter, ListGroup, ListGroupItem } from 'reactstrap';

function User ({ modal, toggle, userdata }) {
  
  const {username, name, email, website, company} = userdata

  return(
    <Modal isOpen={modal} toggle={toggle}>
      <ModalHeader >User details</ModalHeader>
      <ModalBody>
        <ListGroup>
          <ListGroupItem>username - {username}</ListGroupItem>
          <ListGroupItem>fullname - {name}</ListGroupItem>
          <ListGroupItem>email - {email}</ListGroupItem>
          <ListGroupItem>website - {website}</ListGroupItem>
          <ListGroupItem>companyname - {company ? company.name: ''}</ListGroupItem>
        </ListGroup>
      </ModalBody>
      <ModalFooter aria-label="userFooter">
        <Button aria-labelledby="userFooter ok" color="primary" onClick={toggle} tabIndex={1}>Ok</Button>
        <Button aria-labelledby="userFooter cancel" color="secondary" onClick={toggle} tabIndex={1}>Cancel</Button>
      </ModalFooter>
    </Modal>
  )
}

User.propTypes = {
  modal: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  userdata: PropTypes.object.isRequired
}

export default User;
