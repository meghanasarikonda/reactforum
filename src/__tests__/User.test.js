import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import User from '../components/User';

function setup() {
  const props = {
    modal: jest.bool,
    toggle: jest.fn(),
    userdata: {
      username: 'meghana',
      email: 'meghana@mail.com',
      website: 'meghana.dev',
      companydetails: {
        name: 'my company'
      }
    }
  }

  const enzymeWrapper = shallow(<User {...props} />)

  return {
    props,
    enzymeWrapper
  }
}

describe('User', () => {

  it('should render correctly', () => {
    const { enzymeWrapper } = setup()
    expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
  });

  it('should have three children', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.children().length).toBe(3);
  });

  it('shallow wrapper instance should be null', () => {
    const { enzymeWrapper } = setup()
    const instance = enzymeWrapper.instance();

    expect(instance).toBe(null);
  });

  it('should have modal', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.find('Modal').length).toBe(1);
  });

  it('should have modal header', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalHeader').length).toBe(1)
  });

  it('should have modal body', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalBody').length).toBe(1)
  });

  it('should have ListGroupItem', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ListGroupItem').length).toBe(5)
  });

  it('should have modal footer', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalFooter').length).toBe(1)
  });

  it('Modal Footer should have 2 buttons', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('ModalFooter').children().find('Button').length).toBe(2)
  });

  it('Modal should have isOpen as true', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({ modal: true })
    expect(enzymeWrapper.find('Modal').props().isOpen).toBe(true)
  });

  it('Modal should have a false isOpen property', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('Modal').props().isOpen).toBe(false)
  });

})