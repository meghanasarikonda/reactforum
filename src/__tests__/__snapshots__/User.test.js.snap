// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[MockFunction]}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    wrapTag="div"
  >
    User details
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <ListGroup
      tag="ul"
    >
      <ListGroupItem
        tag="li"
      >
        username - 
        meghana
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        fullname - 
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        email - 
        meghana@mail.com
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        website - 
        meghana.dev
      </ListGroupItem>
      <ListGroupItem
        tag="li"
      >
        companyname - 
      </ListGroupItem>
    </ListGroup>
  </ModalBody>
  <ModalFooter
    aria-label="userFooter"
    tag="div"
  >
    <Button
      aria-labelledby="userFooter ok"
      color="primary"
      onClick={[MockFunction]}
      tabIndex={1}
      tag="button"
    >
      Ok
    </Button>
    <Button
      aria-labelledby="userFooter cancel"
      color="secondary"
      onClick={[MockFunction]}
      tabIndex={1}
      tag="button"
    >
      Cancel
    </Button>
  </ModalFooter>
</Modal>
`;
